Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1106,7 +1106,7 @@ namespace Babylon
         const auto bytes{static_cast<uint8_t*>(data.ArrayBuffer().Data()) + data.ByteOffset()};
         if (data.ByteLength() != bimg::imageGetSize(nullptr, width, height, 1, false, false, 1, format))
         {
-            throw std::runtime_error{"The data size does not match width, height, and format"};
+            Napi::Error::New(Env(), "The data size does not match width, height, and format").ThrowAsJavaScriptException();
```

Comment:
this or you need to return to prevent the following code from running
```suggestion
            throw Napi::Error::New(Env(), "The data size does not match width, height, and format");
```